romancal/skycell/match.py

```diff
@@ -0,0 +1,225 @@
+                    bbintervals = iwcs.bounding_box.intervals
+                    # This compensates for the half pixel adjustment in the general code.
+                    image_shape = (
+                        bbintervals[1].upper + 0.5,
+                        bbintervals[0].upper + 0.5,
+                    )
+                elif hasattr(iwcs, "pixel_shape") and iwcs.pixel_shape is not None:
+                    image_shape = (iwcs.pixel_shape[1], iwcs.pixel_shape[0])
+                else:
+                    raise ValueError(
+                        "Cannot infer image footprint from GWCS object because "
+                        "`image_shape` was not specified and "
+                        "the GWCS object does not have `.bounding_box` nor `.pixel_shape`."
+                    )
+
+            iwcs.bounding_box = wcs_bbox_from_shape(image_shape)
+
+        return cls(iwcs.footprint(center=False))
+
+    @property
+    def radec_corners(self) -> NDArray:
+        """corners in right ascension and declination in counterclockwise order"""
+        return self._radec_corners
+
+    @cached_property
+    def radec_center(self) -> tuple[float, float]:
+        """center point in right ascension and declination"""
+        return sgv.vector_to_lonlat(self.vectorpoint_center)
+
+    @cached_property
+    def vectorpoint_corners(self) -> NDArray[float]:
+        """corners in 3D Cartesian space on the unit sphere"""
+        return sgv.normalize_vector(
+            np.stack(sgv.lonlat_to_vector(*np.array(self.radec_corners).T), axis=1)
+        )
+
+    @cached_property
+    def vectorpoint_center(self) -> tuple[float, float, float]:
+        """center in 3D Cartesian space on the unit sphere"""
+        return sgv.normalize_vector(np.mean(self.vectorpoint_corners, axis=0))
+
+    @cached_property
+    def length(self) -> float:
+        """diagonal length of the rectangular footprint"""
+        # assume radial against sky background
+        return sga.length(self.vectorpoint_corners[0], self.vectorpoint_corners[2])
+
+    @cached_property
+    def circumference(self) -> float:
+        """circumference of the rectangular footprint"""
+        return sum(
+            sga.length(
+                self.vectorpoint_corners[index], self.vectorpoint_corners[index + 1]
+            )
+            for index in range(-1, len(self.vectorpoint_corners) - 1)
+        )
+
+    @cached_property
+    def polygon(self) -> sgp.SingleSphericalPolygon:
+        """spherical polygon representing this image footprint"""
+        return sgp.SingleSphericalPolygon(
+            points=self.vectorpoint_corners,
+            inside=self.vectorpoint_center,
+        )
+
+    @cached_property
+    def area(self) -> float:
+        """area of this footprint on the sphere in degrees squared"""
+        return self.polygon.area()
+
+    @cached_property
+    def possibly_intersecting_projregions(self) -> int:
```

Comment:
I wonder if this is unnecessary. All this machinery is hinged on skycells, which are never larger than a projection region.

Reply:
In principle we could define a projection region that way, but we never would.